crates/neon-sys/src/neon.cc

```diff
@@ -177,6 +177,19 @@ extern "C" size_t NeonSys_String_Data(char *out, size_t len, v8::Local<v8::Value
   return Nan::DecodeWrite(out, len, str, Nan::UTF8);
 }
 
+extern "C" bool NeonSys_String_NewFromUCS2(v8::Local<v8::String> *out, v8::Isolate *isolate, const uint16_t *data, int32_t len) {
+  Nan::MaybeLocal<v8::String> maybe = v8::String::NewFromTwoByte(isolate, data, v8::NewStringType::kNormal, len);
+  return maybe.ToLocal(out);
+}
+
+extern "C" int32_t NeonSys_String_UCS2Length(v8::Local<v8::String> str) {
+  return str->Length();
+}
+
+extern "C" size_t NeonSys_String_UCS2Data(uint16_t *out, size_t len, v8::Local<v8::String> str) {
+  return str->Write(out, 0, -1);
```

Comment:
Since you have `len`, I'd pass that in instead of `-1`.  You might want to pass in `NO_NULL_TERMINATION` as well.